File: meetcode.client/src/types/testResult.ts
import { ParsedTestCase, TestCase } from "./testCase";

export interface TestResult {
    testCase: TestCase;
    result: string;
    isSuccessful: boolean;
    execTimeMs: number;
};

export interface ParsedTestResult extends Omit<TestResult, "testCase"> {
    testCase: ParsedTestCase;
}

File: MeetCode.Application/DTOs/Other/TestResult.cs
using MeetCode.Application.DTOs.Response.TestCase;
using MeetCode.Domain.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MeetCode.Application.DTOs.Other
{
    public sealed record TestResult(Guid TestCaseId, string Result, bool IsSuccessful, long ExecTimeMs);
}
